### ISSUE TITLE
Nested routing and spaces in params

### ISSUE THREAD

**Issue by jmorel** [molefrog/wouter#481]
The routing seems to fail when a nested switch encounters a space in a param.

In the following demo app (you can [test it on codesandbox](https://codesandbox.io/p/sandbox/wouter-spaces-n8zst6)), a click on the first 2 links brings out the desired behavior (in the first case it displays `Home` and in the second one it displays `Default page content`). However, with the second set of links, the ones with a space in their param, the switch from the Page component returns absolutely nothing (see screenshot below). It never even fallbacks on the default route.

```ts
import { Switch, Route, Link } from "wouter";

const Page = ({ params: { name } }) => {
  return (
    <>
      <h2>Page {name}</h2>
      <Switch>
        <Route path="/home">Home</Route>
        <Route>Default page content</Route>
      </Switch>
    </>
  );
};

export default function App() {
  return (
    <>
      <ul>
        <li>
          <Link to="~/page/foo/home">foo home</Link>
        </li>
        <li>
          <Link to="~/page/foo">foo default content</Link>
        </li>
        <li>
          <Link to="~/page/foo bar/home">foo bar home</Link>
        </li>
        <li>
          <Link to="~/page/foo bar">foo bar default content</Link>
        </li>
      </ul>
      <Switch>
        <Route path="/page/:name" component={Page} nest />
        <Route>Default route</Route>
      </Switch>
    </>
  );
}
```

![Capture d’écran 2024-08-20 à 15 27 52](https://github.com/user-attachments/assets/d23ea99a-aaf8-42b5-97c0-5925190a0ae1)

**Comment by molefrog** [comment]
Thank you for spotting this. It definitely isn’t the desired behavior. I believe the issue lies in how we’re constructing a relative path using an unescaped base path. I’ve modified the [sandbox](https://codesandbox.io/p/sandbox/wouter-base-security-error-m2939n?from-embed=&workspaceId=48259697-26b1-47bc-9e72-ae2a4a28f7d9&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm03mv2lg00063b6ik58ax9qj%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm03mv2lg00023b6i8kkax928%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm03mv2lg00033b6i2tw75f0y%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm03mv2lg00053b6iymoe47wu%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm03mv2lg00023b6i8kkax928%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm03mv2lg00013b6ix2vg1xon%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%257D%255D%252C%2522id%2522%253A%2522cm03mv2lg00023b6i8kkax928%2522%252C%2522activeTabId%2522%253A%2522cm03mv2lg00013b6ix2vg1xon%2522%257D%252C%2522cm03mv2lg00053b6iymoe47wu%2522%253A%257B%2522id%2522%253A%2522cm03mv2lg00053b6iymoe47wu%2522%252C%2522activeTabId%2522%253A%2522cm03mv4el000p3b6inrfozkl5%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm03mv2lg00043b6iavqfknvi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522cm03mv4el000p3b6inrfozkl5%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522cm03mv2lg00033b6i2tw75f0y%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cm03mv2lg00033b6i2tw75f0y%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) to get to the root cause, and I’ll investigate it further.

**Comment by jmorel** [comment]
Thank you for the reactivity here!

**Comment by molefrog** [comment]
Released the bugfix in v3.3.3
